lesson_13/lesson_13_mihaela_vaida/homework13_mihaela_vaida.py

```diff
@@ -0,0 +1,78 @@
+from math import sqrt
+class PerimeterMixin:
+    def __init__(self):
+        self.sides = None
+
+    def perimeter(self):
+        return f' The perimeter is : {sum(self.sides)}"'
+
+
+class Polygons(PerimeterMixin):
+    def __init__(self, *args):
+        super().__init__()
+        self.sides = args
+        # if len(self.sides) < 3:
+        #   raise Exception('This is not a polygon!')
+        for side_index, lenght in enumerate(self.sides, start=1):
+            if (type(lenght) != int or type(lenght) != float) and lenght <= 0:
+                raise Exception('Invalid lenght !')
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return '{} is the number of sides'.format(no_of_sides)
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print(f'side {side_index} with length: {length}')
+
+
+class Triangle(Polygons):
+    def __init__(self, *args):
+        super().__init__(*args)
+        s1, s2, s3 = self.sides
+        if s1 > s2 + s3 or s2 > s1 + s3 or s3 > s1 + s2:
+            raise Exception("Impossible Triangle!")
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+
+        return sqrt(s_p * (s_p - s1) * (s_p - s2) * (s_p - s3))
+
+
+class Square(Polygons):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def __str__(self):
+        to_be_printed = str()
+        for side_index, length in enumerate(self.sides, start=1):
+            to_be_printed += 'Side {} with length: {:.2f} \n'.format(side_index, length)
+        return to_be_printed
+
+    def area(self):
+        return f' Area is {self.sides[0] ** 2}'
+
+    def from_area(area):
+
+        side = float(f'{sqrt(area):.2f}')
+        if (type(area) == float or type(area) == int) and area > 0:
+
+            print(f'From the area = {area} of a square the side = {side} \n')
+            return Square(area ** 0.5,area ** 0.5, area ** 0.5, area ** 0.5)
+        else:
+            raise Exception('The side cannot be calculated')
+
+    def perimeter(self):
+        return sum(self.sides) * 4
```

Comment:
The task was to do a method for a triangle to do a perimeter.
Also, since you have the Mixin Class in Polygons Class, you have the functionality of the perimeter method.

Reply:
You are right, modified !